lib/src/facts/solaris/virtualization_resolver.cc

```diff
@@ -18,6 +18,17 @@ namespace facter { namespace facts { namespace solaris {
 
     string virtualization_resolver::get_hypervisor(collection& facts)
     {
+        auto isa = facts.get<string_value>(fact::hardware_isa);
+        if (isa && isa->value() == "sparc") {
+            auto ldom_domainrole_control = facts.get<boolean_value>("ldom_domainrole_control");
```

Comment:
That's pretty much the same with all of virtinfo. The most straight-forward way to get this without the `ldom` fact is to call `virtinfo -t` and regex it as below (the plain call to virtinfo doesn't work on Solaris 11 sparc).